projects/soft_999_prajval_malhotra_fw_102/src/main.c

```diff
@@ -0,0 +1,49 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// Counters structure with two counters
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+// function to increment and print counter a 
+void a_plus_one(SoftTimerId timer_id, void *context) {
+  Counters *obj = context;
+  obj->counter_a++;
+  LOG_DEBUG("A = %d\n", obj->counter_a);
+  soft_timer_start_millis(500, a_plus_one, obj, NULL);
```

Comment:
Generally it's a good idea to factor out these constants like 500 into `#define`'d constants with good names so that we know what they represent, and if we have to change them we only have to do so in one place. Make sure you name them something descriptive and suffix them with the unit (in this case `_MS`).